### PR TITLE
Calculate the exact capacity for 2 HashMaps

### PR DESCRIPTION
**Pull request by ljedrz**

### CODE REVIEW

File: src/librustc_driver/profile/trace.rs [hunk 1]
```diff
@@ -220,7 +220,8 @@ pub fn write_counts(count_file: &mut File, counts: &mut HashMap<String,QueryMetr
 }
 
 pub fn write_traces(html_file: &mut File, counts_file: &mut File, traces: &Vec<Rec>) {
-    let mut counts : HashMap<String,QueryMetric> = HashMap::new();
+    let capacity = traces.iter().fold(0, |acc, t| acc + 1 + t.extent.len());
+    let mut counts : HashMap<String,QueryMetric> = HashMap::with_capacity(capacity);
```
Comment (estebank) [inline src/librustc_driver/profile/trace.rs]: Missing comma between `String` and `QueryMetric`.

**Review by estebank — APPROVED** [review]

**Comment by estebank** [comment]
@ljedrz if you have time, please fix the formatting nitpick, in the meantime, I'm approving as is.

@bors r+ rollup

**Comment by ljedrz** [comment]
No problem, done.

File: src/bootstrap/metadata.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ pub fn build(build: &mut Build) {
     build_krate("", build, &mut resolves, "src/libtest");
     build_krate(&build.rustc_features(), build, &mut resolves, "src/rustc");
 
-    let mut id2name = HashMap::new();
+    let mut id2name = HashMap::with_capacity(build.crates.len());
     for (name, krate) in build.crates.iter() {
         id2name.insert(krate.id.clone(), name.clone());
     }
```
Comment (nnethercote) [inline src/bootstrap/metadata.rs]: For what it's worth, this one could also be done with `map()` followed by `collect()`.

**Pull request merged**